tests/integration/nm/ovs_test.py

```diff
@@ -378,6 +378,102 @@ def ovs_bridge_over_bond_system_iface_with_same_name(eth1_up, eth2_up):
     time.sleep(1)
 
 
+@pytest.fixture
+def ovs_bridge_first_and_ovs_interface_with_same_name_ipv4():
+    # The order on this function is important. The OVS bridge must be defined
+    # before the OVS interface.
+    cmdlib.exec_cmd(
+        "nmcli connection add type ovs-port conn.interface br-ex master br-ex "
+        "con-name ovs-port-br-ex".split(),
+        check=True,
+    )
+    cmdlib.exec_cmd(
+        "nmcli connection add type ovs-bridge con-name br-ex conn.interface "
+        "br-ex".split(),
+        check=True,
+    )
+    nmcli_ovs_interface = (
+        "nmcli",
+        "connection",
+        "add",
+        "type",
+        "ovs-interface",
+        "slave-type",
+        "ovs-port",
+        "conn.interface",
+        "br-ex",
+        "master",
+        "ovs-port-br-ex",
+        "con-name",
+        "ovs-if-br-ex",
+        "ipv4.method",
+        "manual",
+        "ipv4.addr",
+        "192.0.2.2/24",
+        "ipv4.dns",
+        "192.0.2.1",
+        "ipv4.routes",
+        "0.0.0.0/0 192.0.2.1",
+    )
+    cmdlib.exec_cmd(nmcli_ovs_interface, check=True)
+    # Wait a little bit for NM to activate above interfaces to do not hit race
+    # problems.
+    time.sleep(1)
+    yield
+    cmdlib.exec_cmd(
+        "nmcli connection del ovs-port-br-ex br-ex ovs-if-br-ex".split(),
+        check=True,
+    )
+
+
+@pytest.fixture
+def ovs_interface_first_and_ovs_bridge_with_same_name_ipv4():
+    # The order on this function is important. The OVS interface must be
+    # defined before the OVS bridge.
+    cmdlib.exec_cmd(
+        "nmcli connection add type ovs-port conn.interface br-ex master br-ex "
+        "con-name ovs-port-br-ex".split(),
+        check=True,
+    )
+    nmcli_ovs_interface = (
```

Comment:
seems like these should be functions now

Reply:
The ovs interface is now a function. I think for the other it is not necessary.